addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/AuthUtils.java

```diff
@@ -533,6 +555,31 @@ static User getUser(Context context, String userName) {
         return oUser.get();
     }
 
+    public static VerificationRequestDetails getVerificationDetailsForContext(int contextId) {
+        return contextVerifMap.get(contextId);
+    }
+
+    public static void setVerificationDetailsForContext(
+            int contextId, VerificationRequestDetails details) {
+        contextVerifMap.put(contextId, details);
+    }
+
+    private static ExecutorService executorService;
+
+    private static ExecutorService getExecutorService() {
```

Comment:
This will be called by the scan rule which might run concurrently, potentially causing more than one executor to be created. (The maps will also have concurrent accesses.)